mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -0,0 +1,499 @@
+
+    # Load mesh related variables
+    try:
+        restartFile = streams.readpath('restart')[0]
+    except ValueError:
+        raise IOError('No MPAS-O restart file found: need at least one '
+                      'restart file for MOC calculation')
+    ncFile = netCDF4.Dataset(restartFile, mode='r')
+    dvEdge = ncFile.variables['dvEdge'][:]
+    areaCell = ncFile.variables['areaCell'][:]
+    refBottomDepth = ncFile.variables['refBottomDepth'][:]
+    latCell = ncFile.variables['latCell'][:]
+    latCell = latCell * rad_to_deg  # convert to degree
+    ncFile.close()
+
+    nVertLevels = len(refBottomDepth)
+    refTopDepth = np.zeros(nVertLevels+1)
+    refTopDepth[1:nVertLevels+1] = refBottomDepth[0:nVertLevels]
+    refLayerThickness = np.zeros(nVertLevels)
+    refLayerThickness[0] = refBottomDepth[0]
+    refLayerThickness[1:nVertLevels] = (refBottomDepth[1:nVertLevels] -
+                                        refBottomDepth[0:nVertLevels-1])
+
+    variableList = ['avgNormalVelocity',
+                    'avgVertVelocityTop']
+
+    # Load basin region related variables and save them to dictionary
+    # NB: The following will need to change with new regional mapping files
+    regionMaskFiles = config.get(sectionName, 'regionMaskFiles')
+    if not os.path.exists(regionMaskFiles):
+        raise IOError('Regional masking file for MOC calculation '
+                      'does not exist')
+    iRegion = 0
+    for region in regionNames:
+        print '\n  Reading region and transect mask for {}...'.format(region)
+        ncFileRegional = netCDF4.Dataset(regionMaskFiles, mode='r')
+        maxEdgesInTransect = ncFileRegional.dimensions['maxEdgesInTransect'].size
+        transectEdgeMaskSigns = ncFileRegional.variables['transectEdgeMaskSigns'][:, iRegion]
+        transectEdgeGlobalIDs = ncFileRegional.variables['transectEdgeGlobalIDs'][iRegion, :]
+        regionCellMask = ncFileRegional.variables['regionCellMasks'][:, iRegion]
+        ncFileRegional.close()
+        iRegion += 1
+
+        indRegion = np.where(regionCellMask == 1)
+        dictRegion = {
+            'ind{}'.format(region): indRegion,
+            '{}CellMask'.format(region): regionCellMask,
+            'maxEdgesInTransect{}'.format(region): maxEdgesInTransect,
+            'transectEdgeMaskSigns{}'.format(region): transectEdgeMaskSigns,
+            'transectEdgeGlobalIDs{}'.format(region): transectEdgeGlobalIDs}
+    # Add Global regionCellMask=1 everywhere to make the algorithm
+    # for the global moc similar to that of the regional moc
+    dictRegion['GlobalCellMask'] = np.ones(np.size(latCell))
+    regionNames[:0] = ['Global']
+
+    # Compute and plot annual climatology of MOC streamfunction
+    print '\n  Compute and/or plot post-processed MOC climatological '\
+          'streamfunction...'
+    simulationStartTime = get_simulation_start_time(streams)
+    outputDirectory = buildConfigFullPath(config, 'output',
```

Comment:
This output directory is appropriate for the streamfunction but maybe it's kind of confusing to write files for the time series here?  Should we make a separate directory for writing out time series data?  We could also cache extractions of time series from other analysis tasks.